beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -344,53 +365,159 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
+                        | ExecutionPayloadError::InvalidTerminalBlockHash { .. } => {
+                            ErrorCategory::malicious_no_retry()
+                        }
+                        // Do not penalize the peer since it's not their fault that *we're* optimistic.
+                        ExecutionPayloadError::UnverifiedNonOptimisticCandidate => {
+                            ErrorCategory::internal_retry()
+                        }
+                    }
+                    .into(),
+                    BlockError::ParentExecutionPayloadInvalid { .. } => {
+                        ErrorCategory::malicious_no_retry().into()
+                    }
+                    // TODO: Review AvailabilityCheckError variants
+                    BlockError::AvailabilityCheck(e) => match e {
+                        AvailabilityCheckError::SszTypes(_)
+                        | AvailabilityCheckError::StoreError(_)
+                        | AvailabilityCheckError::Unexpected
+                        | AvailabilityCheckError::ParentStateMissing(_)
+                        | AvailabilityCheckError::BlockReplayError(_)
+                        | AvailabilityCheckError::RebuildingStateCaches(_)
+                        | AvailabilityCheckError::SlotClockError => ErrorCategory::internal_retry(),
+                        AvailabilityCheckError::InvalidColumn(index, _) => {
+                            ErrorCategory::Malicious {
+                                retry: true,
+                                index: index as usize,
+                            }
+                        }
+                        AvailabilityCheckError::InvalidBlobs { .. }
+                        | AvailabilityCheckError::MissingBlobs
+                        | AvailabilityCheckError::MissingCustodyColumns
+                        | AvailabilityCheckError::DecodeError(_)
+                        | AvailabilityCheckError::ReconstructColumnsError { .. }
+                        | AvailabilityCheckError::BlobIndexInvalid(_)
+                        | AvailabilityCheckError::DataColumnIndexInvalid(_)
+                        | AvailabilityCheckError::KzgCommitmentMismatch { .. } => {
+                            ErrorCategory::malicious_retry()
+                        } // Do not use a fallback match, handle all errors explicitly
+                    }
+                    .into(),
+                    // The proposer making a slashable block is not the peer's fault nor ours. Mark
+                    // as internal (don't penalize peer), and no retry (the block will forever be
+                    // slashable).
+                    BlockError::Slashable => ErrorCategory::internal_no_retry().into(),
+                    // TODO: BeaconChainError should be retried?
+                    BlockError::BeaconChainError(_) | BlockError::InternalError(_) => {
+                        ErrorCategory::internal_no_retry().into()
```

Comment:
Both of these variants cover quite a few scenarios, I'm not sure if it's safe to say no retry to them?